static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -218,14 +285,30 @@ class SpanTreeModel {
           ...span.getSpansList({
             operationNameFilters,
             generateBounds,
-            treeDepth: treeDepth + 1,
+            treeDepth: shouldHideSpanOfGroup ? treeDepth : treeDepth + 1,
             isLastSibling: index === lastIndex,
             continuingTreeDepths: descendantContinuingTreeDepths,
             hiddenSpanGroups,
             spanGroups: new Set(childSpanGroup),
             filterSpans,
             previousSiblingEndTimestamp: acc.previousSiblingEndTimestamp,
             event,
+            isOnlySibling: descendantsSource.length === 1,
+            spanGrouping: shouldGroup
+              ? [...(spanGrouping ?? []), wrappedSpan]
+              : undefined,
+            toggleSpanGroup:
+              toggleSpanGroup === undefined && isNotLastSpanOfGroup
+                ? this.toggleSpanGroup
+                : toggleSpanGroup !== undefined && isNotLastSpanOfGroup
+                ? toggleSpanGroup
+                : undefined,
+            showSpanGroup:
+              toggleSpanGroup === undefined && isNotLastSpanOfGroup
+                ? this.showSpanGroup
+                : toggleSpanGroup !== undefined && isNotLastSpanOfGroup
+                ? showSpanGroup
+                : false,
```

Comment:
I think we could reduce verbosity with something like this:
```
            toggleSpanGroup: isNotLastSpanOfGroup
              ? toggleSpanGroup === undefined
                ? this.toggleSpanGroup
                : toggleSpanGroup
              : undefined,
            showSpanGroup: isNotLastSpanOfGroup
              ? toggleSpanGroup === undefined
                ? this.showSpanGroup
                : showSpanGroup
              : false,
```
Hopefully this looks more clear, ignore if not lol

Reply:
Done.